src/blr/cli/tagfastq.py

```diff
@@ -0,0 +1,132 @@
+
+    reads_skipped = list()
+    # Parse input FASTA/FASTQ for read1 and read2 and write output
+    with dnaio.open(args.input1, file2=args.input2, interleaved=in_interleaved, mode="r") as reader, \
+            dnaio.open(args.output1, file2=args.output2, interleaved=out_interleaved, mode="w") as writer:
+        for read1, read2 in tqdm(reader, desc='Read pairs processed'):
+            # Header parsing
+            name_and_pos_r1, read_and_index_r1 = read1.name.split(maxsplit=1)
+            name_and_pos_r2, read_and_index_r2 = read2.name.split(maxsplit=1)
+
+            assert name_and_pos_r1 == name_and_pos_r2
+
+            try:
+                raw_barcode_seq = header_barcodes[name_and_pos_r1]
+            except KeyError:
+                reads_skipped.append(name_and_pos_r1)
+                continue
+
+            corr_barcode = corrected_barcodes[raw_barcode_seq]
+
+            # Make new string with barcode sequence and id to add to headers.
+            new_text = f"{corr_barcode.seq}_{args.barcode_cluster_tag}:Z:{corr_barcode.id}"
+
+            # Save header to read instances
+            read1.name = f"{name_and_pos_r1}_{new_text} {read_and_index_r1}"
+            read2.name = f"{name_and_pos_r2}_{new_text} {read_and_index_r2}"
+
+            # Write to out
+            writer.write(read1, read2)
+
+    for header in reads_skipped:
+        logger.info(f"Barcode for read-pair {header} not found. Skipped read-pair.")
+
+    logger.info(f'Finished')
+
+
+def parse_corrected_barcodes(open_file):
+    """
+    Parse starcode cluster output and return a dictionary with raw sequences pointing to a
+    corrected canonical sequence
+    :param open_file: starcode tabular output file.
+    :return: dict: raw sequences pointing to a namedtuple containing corrected canonical sequence and id.
+    """
+    target = namedtuple("Cluster", ['seq', 'id'])
+    corrected_barcodes = dict()
+    for id, cluster in tqdm(enumerate(open_file.readlines(), start=1), desc="Corrected barcodes processed"):
+        canonical_seq, _, cluster_seqs = cluster.strip().split("\t", maxsplit=3)
+        cluster_target = target(seq=canonical_seq, id=id)
+        corrected_barcodes.update({raw_seq: cluster_target for raw_seq in cluster_seqs.split(",")})
+    return corrected_barcodes
+
+
+def parser_raw_barcodes(open_file):
+    """
+    Parse FASTA/FASTQ containing barcode sequences and return a dictionary with entry headers pointing to a
+    raw barcodes sequence.
+    :param open_file: dnaio odject.
+    :return: dict: entry headers pointing to a raw barcodes sequence
+    """
+    header_barcodes = dict()
+    for barcode in tqdm(open_file, desc="Raw barcodes processed"):
+        header, _ = barcode.name.split(maxsplit=1)
+        header_barcodes[header] = barcode.sequence
+    return header_barcodes
+
+
+def add_arguments(parser):
+    parser.add_argument(
+        "input1", metavar='<INPUT1>',
```

Comment:
Please leave out the `metavar=` here and in all places, where only the angled brackets are added. They don’t make the help output more readable.

In general, if you want behavior different from the argparse default, you should change how argparse constructs the help string, not adjust the string in each call to `add_argument`.